test/test-platform-output.c

```diff
@@ -134,6 +142,98 @@ TEST_IMPL(platform_output) {
+      } else if (network_interfaces[i].address.address4.sin_family == AF_INET) {
+        uv_ip4_name(&network_interfaces[i].address.address4,
+                    buffer, sizeof(buffer));
+        printf("    address: inet4 %s\n", buffer);
+      } else {
+        printf("    address: none\n");
+      }
+
+      /* Broadcast */
+      if (network_interfaces[i].broadcast.broadcast4.sin_family == AF_INET6) {
+        uv_ip6_name(&network_interfaces[i].broadcast.broadcast6,
+                    buffer, sizeof(buffer));
+        printf("    broadcast: inet6 %s\n", buffer);
+      } else if (network_interfaces[i].broadcast.broadcast4.sin_family == AF_INET) {
+        uv_ip4_name(&network_interfaces[i].broadcast.broadcast4,
+                    buffer, sizeof(buffer));
+        printf("    broadcast: inet4 %s\n", buffer);
+      } else {
+        printf("    broadcast: none\n");
+      }
+
+      /* Netmask */
+      if (network_interfaces[i].netmask.netmask4.sin_family == AF_INET6) {
+        uv_ip6_name(&network_interfaces[i].netmask.netmask6,
+                    buffer, sizeof(buffer));
+        printf("    netmask: inet6 %s\n", buffer);
+      } else if (network_interfaces[i].netmask.netmask4.sin_family == AF_INET) {
+        uv_ip4_name(&network_interfaces[i].netmask.netmask4,
+                    buffer, sizeof(buffer));
+        printf("    netmask: inet4 %s\n", buffer);
+      } else {
+        printf("    netmask: none\n");
+      }
+
+      /* Physical layer */
+      ASSERT(network_interfaces[i].phys_addr_len == 0 ||
+             network_interfaces[i].phys_addr_len == 6 ||
+             network_interfaces[i].phys_addr_len == 8);
+      if (network_interfaces[i].phys_addr_len == 6) {
+        printf("    physical address: "
+               "%02x:%02x:%02x:%02x:%02x:%02x\n",
+               (unsigned char)network_interfaces[i].phys_addr[0],
+               (unsigned char)network_interfaces[i].phys_addr[1],
+               (unsigned char)network_interfaces[i].phys_addr[2],
+               (unsigned char)network_interfaces[i].phys_addr[3],
+               (unsigned char)network_interfaces[i].phys_addr[4],
+               (unsigned char)network_interfaces[i].phys_addr[5]);
+      } else if (network_interfaces[i].phys_addr_len == 8) {
+        printf("    physical address: "
+               "%02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x\n",
+               (unsigned char)network_interfaces[i].phys_addr[0],
+               (unsigned char)network_interfaces[i].phys_addr[1],
+               (unsigned char)network_interfaces[i].phys_addr[2],
+               (unsigned char)network_interfaces[i].phys_addr[3],
+               (unsigned char)network_interfaces[i].phys_addr[4],
+               (unsigned char)network_interfaces[i].phys_addr[5],
+               (unsigned char)network_interfaces[i].phys_addr[6],
+               (unsigned char)network_interfaces[i].phys_addr[7]);
+      } else {
+        printf("    physical address: none\n");
```

Comment:
`ASSERT(network_interfaces[i].phys_addr_len == 0)`?

Reply:
> `ASSERT(network_interfaces[i].phys_addr_len == 0)?`

@bnoordhuis That would fail on `tun` interfaces, wouldn't it?